### PR TITLE
Stricter linting and enforcing

### PR DESCRIPTION
**Pull request by madarche**

### CODE REVIEW

**Comment by madarche** [comment]
@zaach could you merge this PR too please? Thank you!

**Comment by madarche** [comment]
@vladikoff since you're around, could you check this one too please? Thanks.

File: lib/convict.js [hunk 1]
```diff
@@ -293,6 +293,8 @@ function coerce(k, v, schema, instance) {
         v = moment.duration(parseInt(split[0], 10), split[1]).valueOf();
       }
       break;
+    default:
+        // TODO: Should we throw an exception here?
```
Comment (vladikoff) [inline lib/convict.js]: I guess we can fix that in the future?

Comment (madarche) [inline lib/convict.js]: Right, I didn't want to modify the behavior with this commit, I just wanted to set up "protections" for future commits.

Comment (vladikoff) [inline lib/convict.js]: :+1:

**Comment by madarche** [comment]
Thanks again @vladikoff!

**Comment by vladikoff** [comment]
@madarche no thank you! :+1:

**Pull request merged**